src/OpenTelemetry.Exporter.Geneva/Metrics/OtlpProtobuf/ProtobufSerializerHelper.cs

```diff
@@ -0,0 +1,233 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Buffers.Binary;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace OpenTelemetry.Exporter.Geneva;
+
+internal static class ProtobufSerializerHelper
+{
+    private const int Fixed64Size = 8;
+
+    internal static Encoding Utf8Encoding => Encoding.UTF8;
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    internal static void WriteStringTag(byte[] buffer, ref int currentPosition, string value, int fieldNumber)
+    {
+        int stringSize = Utf8Encoding.GetByteCount(value);
+
+        WriteTag(buffer, ref currentPosition, fieldNumber, WireFormat.WireType.LengthDelimited);
+
+        WriteLength(buffer, ref currentPosition, stringSize);
+
+        _ = Encoding.UTF8.GetBytes(value, 0, value.Length, buffer, currentPosition);
```

Comment:
This returns the size of string, is there a way to update the buffer with value and then go back and update the length?
There were some todo's not sure if it was to sort this out.

Reply:
This is different than other cases. Here we want to optimize on the number on bytes used to encode length. By writing the length first we ensure that we used the minimum # of required bytes.